src/hooks/useBootstrapApp.ts

```diff
@@ -0,0 +1,49 @@
+import { useQuery } from 'react-query';
+
+import { CACHE_TIME, STALE_TIME } from '#src/config';
+import { getModule } from '#src/container';
+import AppController from '#src/stores/AppController';
+import { useTrackConfigKeyChange } from '#src/hooks/useTrackConfigKeyChange';
+import { SettingsController } from '#src/stores/SettingsController';
+import type { Config } from '#types/Config';
+import type { Settings } from '#src/stores/SettingsStore';
+
+const applicationController = getModule(AppController);
+const settingsController = getModule(SettingsController);
+
+type Resources = {
+  config: Config;
+  configSource: string | undefined;
+  settings: Settings;
+};
+
+export const useBootstrapApp = (onReady: () => void) => {
+  const { data, isLoading, error, isError, isSuccess } = useQuery<Resources, Error>(
+    'config-init',
+    async () => {
+      const settings = await settingsController.initialize();
+      const configSource = settingsController.getConfigSource(settings);
+      const config = await applicationController.initApp(configSource);
```

Comment:
Can you combine these further? AppController already depends on a config service, but can also depend on the settings service. Actually config service could potentially depend on the settings service, so if you want to build one that doesn't care about the settings concept you could.

Reply:
Yes, we could! Then the AppController would also make more sense as it orchestrates more than just the config logic.

Reply:
I've updated the PR with the changes proposed. What do you think?

Reply:
Nice! Looks like we can also make SettingsService optional, right?

Reply:
We could, but the SettingsService is now responsible for determining the config id (`configSource`). We would need to refactor that. Do you have suggestions? 😄

Thinking about this, we could swap the `SettingsService` for production builds with a static service. For example;

```ts
class DemoSettingsService extends SettingsService {
  // current SettingsService with URL search param logic
}

class SettingsService {
  initialize = async () {
    return {
      defaultConfigSource: import.meta.env.APP_DEFAULT_CONFIG_SOURCE,
      playerId: import.meta.env.APP_PLAYER_ID || OTT_GLOBAL_PLAYER_ID,
      playerLicenseKey: import.meta.env.APP_PLAYER_LICENSE_KEY,
    };
  }

  getConfigSource(settings: Settings | undefined) {
    if (!settings) {
      return '';
    }

    return settings?.defaultConfigSource;
  }
}
```

But that's for later I guess 😅